### PR TITLE
Update previous Crystal release - 1.4.1

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -52,13 +52,13 @@ let
   # Hashes obtained using `nix-prefetch-url --unpack <url>`
   latestCrystalBinary = genericBinary ({
     x86_64-darwin = {
-      url = "https://github.com/crystal-lang/crystal/releases/download/1.3.2/crystal-1.3.2-1-darwin-universal.tar.gz";
-      sha256 = "sha256:1jxc5c82wxigza2z0ck51dnwpvfzba0gsczrjxgfjysqn6ki7v3q";
+      url = "https://github.com/crystal-lang/crystal/releases/download/1.4.1/crystal-1.4.1-1-darwin-universal.tar.gz";
+      sha256 = "sha256:0ckbsgr1y64w0yj80dn8v7zi6zqdlpin35l742rrblhzbdhdkalw";
```
Comment (beta-ziliani) [inline shell.nix:56]:
```suggestion
      sha256 = "sha256:e7f9b3e1e866dc909a0a310238907182f1ee8b3c09bd8da5ecd0072d99c1fc5c";
```

File: shell.nix [hunk 2]
```diff
@@ -52,13 +52,13 @@ let
...
     };
 
     x86_64-linux = {
-      url = "https://github.com/crystal-lang/crystal/releases/download/1.3.2/crystal-1.3.2-1-darwin-universal.tar.gz";
-      sha256 = "sha256:1jxc5c82wxigza2z0ck51dnwpvfzba0gsczrjxgfjysqn6ki7v3q";
+      url = "https://github.com/crystal-lang/crystal/releases/download/1.4.1/crystal-1.4.1-1-linux-x86_64.tar.gz";
+      sha256 = "sha256:1j5b2f4s2g8c1x7mhvsdzzjw878x0kkbrih4plxlzg3hmijvrnwj";
```
Comment (beta-ziliani) [inline shell.nix:61]:
```suggestion
      sha256 = "sha256:a5bdf1b78897b3cdc7d715b5f7adff79e84401d39b7ab546ab3249dc17fc770c";
```

**Review by beta-ziliani — CHANGES_REQUESTED** [review]
The sha256 seems to be wrong, pls double check.

**Comment by straight-shoota** [comment]
@beta-ziliani I can't test darwin, but linux definitely works for me.
Reminder: https://github.com/crystal-lang/crystal/pull/11715#discussion_r789057526

**Comment by straight-shoota** [comment]
Well, if darwin failed, we wouldn't see a green CI because nix shell is actually used in `x86_64-darwin-test`.

**Pull request merged**